hcloud/action.go

```diff
@@ -224,20 +225,27 @@ func (c *ActionClient) WatchOverallProgress(ctx context.Context, actions []*Acti
 				return
 			}
 
+			progress := 0
 			for _, a := range as {
 				switch a.Status {
 				case ActionStatusRunning:
-					continue
+					progress += a.Progress
 				case ActionStatusSuccess:
 					delete(watchIDs, a.ID)
-					successIDs = append(successIDs, a.ID)
-					sendProgress(progressCh, int(float64(len(actions)-len(successIDs))/float64(len(actions))*100))
+					completedIDs = append(completedIDs, a.ID)
 				case ActionStatusError:
 					delete(watchIDs, a.ID)
+					completedIDs = append(completedIDs, a.ID)
 					errCh <- fmt.Errorf("action %d failed: %w", a.ID, a.Error())
 				}
 			}
 
+			progress += (len(completedIDs) * 100)
+			if progress != 0 && progress != previousProgress {
+				sendProgress(progressCh, int(progress/len(actions)))
```

Comment:
`progress / len(actions)` has the potential to panic if the user passes 0 actions to this method. Not an issue right now because we have a check `len(as)==0 { return }` but good to be mindful of this.